crates/bevy_core_pipeline/src/bloom/downsampling_pipeline.rs

```diff
@@ -0,0 +1,197 @@
+use bevy_ecs::{
+    prelude::{Component, Entity},
+    query::With,
+    system::{Commands, Query, Res, ResMut, Resource},
+    world::{FromWorld, World},
+};
+use bevy_math::Vec4;
+use bevy_render::{
+    render_resource::{
+        BindGroupLayout, BindGroupLayoutDescriptor, BindGroupLayoutEntry, BindingType,
+        BufferBindingType, CachedRenderPipelineId, ColorTargetState, ColorWrites, FragmentState,
+        MultisampleState, PipelineCache, PrimitiveState, RenderPipelineDescriptor,
+        SamplerBindingType, Shader, ShaderStages, ShaderType, SpecializedRenderPipeline,
+        SpecializedRenderPipelines, TextureSampleType, TextureViewDimension,
+    },
+    renderer::RenderDevice,
+    view::ExtractedView,
+};
+
+use crate::fullscreen_vertex_shader::fullscreen_shader_vertex_state;
+
+use super::{BloomSettings, BLOOM_SHADER_HANDLE, BLOOM_TEXTURE_FORMAT};
+
+#[derive(Component)]
+pub struct BloomDownsamplingPipelineIds {
+    pub id_main: CachedRenderPipelineId,
```

Comment:
Remove `id` from the fields, the struct name already indicates that, e.g. `bloom_downsampling_pipeline_ids.main/first`